### DISCUSSION TITLE
Automating spacing values specific to screen size

### DISCUSSION THREAD

**Discussion by brydonm** [tailwindlabs/tailwindcss#12661] [Help] [answered]
Hey everyone. Wondering if anyone would be able to suggest an idea on automating screen size variants to config classes. I'd like to avoid developers needing to use `m-sm-mobile lg:m-sm` and instead use `m-sm`

```
  theme: {
    extend: {
      spacing: {
        '2xs': '0.313rem',
        xs: '0.625rem',
        sm: '1.25rem',
        'sm-mobile': '0.938rem',
        md: '1.875rem',
        'md-mobile': '1.25rem',
        lg: '2.813rem',
        'lg-mobile': '1.875rem',
        xl: '3.75rem',
        'xl-mobile': '2.5rem',
        '2xl': '5rem',
        '2xl-mobile': '3.75rem',
        '3xl': '7.5rem',
        '3xl-mobile': '5rem',
      },
    },
  },
```

For example, `m-sm` should apply `m-sm-mobile lg:m-sm`. I tried writing a utility for this but didn't have much luck applying classes that already exist.

**Comment by wongjn** [comment, marked as answer]
You could:

- Create your own plugin [like this](https://play.tailwindcss.com/caCDgEEDpi?file=config).
- Use CSS variables [like this](https://play.tailwindcss.com/w9tUZgMYYg?file=css).
- Use your templating engine to store variables that contain the class strings, like:
  ```twig
  {% set m_sm = 'm-sm-mobile lg:m-sm' %}
  {# … #}
  <div class="{{ m_sm }}">
  ```

**Reply by brydonm** [reply]
Thanks for the ideas @wongjn . I think CSS variables would be the easiest approach here!
